streamz/core.py

```diff
@@ -1231,6 +1236,52 @@ def cb(self):
             yield self._emit(x)
 
 
+@Stream.register_api()
+class to_kafka(Stream):
+    """ Writes data in the stream to Kafka
+
+    The Confluent client sends messages in batches and asynchronously. When a
+    message is received, or in the event of an error, this stream will emit
+    a tuple in the form of (err, msg) see
+    https://github.com/confluentinc/confluent-kafka-python
+
+    Parameters
+    ----------
+    topic : string
+        The topic which to write
+    producer_config : dict
+        Settings to set up the stream, see
+        https://docs.confluent.io/current/clients/confluent-kafka-python/#configuration
+        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        Examples:
+        bootstrap.servers: Connection string(s) (host:port) by which to reach Kafka
+
+    Examples
+    --------
+    >>> source = Stream()
+    >>> source.map(lambda x: x + 1).to_kafka('test', {'bootstrap.servers': 'localhost'}).sink(
+    ...     lambda x: print(x[1].topic(), x[1].partition()))
+    >>> for i in range(3):
+    ...     source.emit(i)
+    test 0
+    test 0
+    test 0
+    """
+    def __init__(self, upstream, topic, producer_config, **kwargs):
+        self.topic = topic
+        self.producer = ck.Producer(producer_config)
+        atexit.register(self.producer.flush)
+
+        Stream.__init__(self, upstream, ensure_io_loop=True, **kwargs)
+
+    def update(self, x, who=None):
+        self.producer.poll(0)
+        self.producer.produce(self.topic, str(x).encode('utf-8'), callback=self.cb)
```

Comment:
I'm not sure casting to string is desired here, I believe the end user should create an object that is either bytes-like or str-like to be able to send and you should just send the object. The string representation of an object could be drastically different than the byte representation of the object.

Reply:
You probably do want to ensure that the thing passed is bytes-like, I'm not sure I would trust confluent's library not to segfault... (perhaps make a test for something not bytes-like to make sure it behaves?)

Reply:
So, if the user is manipulating text data, then they have to to have a stage upstream that will do the bytes encoding?

Reply:
This change seems to work.

https://github.com/jsmaupin/streamz/commit/893300f55e8d016b91e9770471cc1109dd34a742

Reply:
I have updated the PR.  The Confluent library will take in a string or bytes, so Streamz will pass in the user provided value directly to the Confluent library

Reply:
I'm glad to hear it.  Thanks for tracking this down @jsmaupin